sys/fmt/fmt.c

```diff
@@ -251,48 +251,56 @@ size_t fmt_s16_dec(char *out, int16_t val)
     return fmt_s32_dec(out, val);
 }
 
-size_t fmt_s16_dfp(char *out, int16_t val, unsigned fp_digits)
+size_t fmt_s16_dfp(char *out, int16_t val, int fp_digits)
 {
     return fmt_s32_dfp(out, val, fp_digits);
 }
 
-size_t fmt_s32_dfp(char *out, int32_t val, unsigned fp_digits)
+size_t fmt_s32_dfp(char *out, int32_t val, int fp_digits)
 {
-    assert(fp_digits < TENMAP_SIZE);
+    assert(fp_digits > -(int)TENMAP_SIZE);
 
-    int32_t absolute, divider;
-    unsigned div_len, len, pos = 0;
-    char tmp[9];
+    unsigned  pos = 0;
 
     if (fp_digits == 0) {
-        return fmt_s32_dec(out, val);
+        pos = fmt_s32_dec(out, val);
     }
-    if (val < 0) {
+    else if (fp_digits > 0) {
+        pos = fmt_s32_dec(out, val);
         if (out) {
-            out[pos++] = '-';
+            memset(&out[pos], '0', fp_digits);
         }
-        val = -val;
-    }
-
-    uint32_t e = _tenmap[fp_digits];
-    absolute = (val / e);
-    divider = val - (absolute * e);
-
-    pos += fmt_s32_dec(&out[pos], absolute);
-
-    if (!out) {
-        return pos + 1 + fp_digits;     /* abs len + decimal point + divider */
+        pos += fp_digits;
     }
+    else {
+        fp_digits *= -1;
+        uint32_t e = _tenmap[fp_digits];
+        int32_t abs = (val / (int32_t)e);
+        int32_t div = val - (abs * e);
+
+        /* the divisor should never be negative */
+        if (div < 0) {
+            div *= -1;
```

Comment:
IIRC @gebart suggested (or showed) that it is also possible to write `div = -div;`, but that's no blocker.

Reply:
`div *= -1` seems to me more intuitive to me and code size is identical, so I tend to prefer this style. But no hard feeling about, if consensus is to go for `x = -x` in general, I will fix this...

Reply:
I don't care for this PR. In general, keep the style of the edited file, which is "x=-x".